### DISCUSSION TITLE
How can I use the new Dataset API to delete a column/metric?

### DISCUSSION THREAD

**Discussion by etr2460** [apache/superset#12819] [Q&A / Help]
See question in the title. The only API i see for Datasets is `PUT /api/v1/dataset/<pk>`, which doesn't seem to support deleting a column or a metric. Is this supported/planned to be supported in the future?

**Comment by willbarrett** [comment]
@dpgaspar is this something that was missed on initial implementation, or is there some FAB magic we're missing?

**Comment by dpgaspar** [comment]
@etr2460 you are right, and there is no missing magic @willbarrett. It seems to be missing, the columns and metrics API chunk is not used by the frontend yet, but I think it's something we will support in the future (no defined dates yet)

**Reply by etr2460** [reply]
Gotcha, thanks for the update! For context, I ended up here because we currently use the old FAB api for updating/deleting columns and metrics, and it's unreasonably slow for some actions. Perhaps I should file a bug against FAB for now then? Let me know what you think @dpgaspar 

<img width="1334" alt="Screen Shot 2021-01-27 at 10 13 05 AM" src="https://user-images.githubusercontent.com/7409244/106840203-89326780-6654-11eb-9acd-0adc1dc79dd7.png">

**Reply by dpgaspar** [reply]
@etr2460 I would prefer an issue on Superset, I think it's better not to waste cycles supporting the old API.
